Formula/synergy-core.rb

```diff
@@ -0,0 +1,351 @@
+      ohai "Thus, it is not possible to do any nontrivial testing of synergy-core."
+      ohai ""
+      ohai "This is not considered a test failure because it does not indicate any problem"
+      ohai "with brew's installation of synergy-core."
+    end
+
+    def kill_and_wait_for(process)
+      Process.kill("KILL", process.pid)
+      Process.wait(process.pid)
+    end
+
+    # In certain cases (discussed below), it is not possible to do any nontrivial
+    # tests. To make sure that Synergy at least compiled successfully, we'll do
+    # some totally trivial checks here. These checks are bad, but they are better
+    # than nothing, which is what would otherwise be possible in certain cases.
+    assert_equal `#{opt_bin}/synergys`, "synergys: no configuration available\n"
+    assert_equal `#{opt_bin}/synergyc`.split("\n")[0], "synergyc: a server address or name is required"
+    synergys_version = `#{opt_bin}/synergys --version`.split("\n")[0]
+    synergyc_version = `#{opt_bin}/synergyc --version`.split("\n")[0]
+    if build.head?
+      version_string = Regexp.quote(version.to_s.split("-")[1])
+      assert_match(/synergys [\-.0-9a-z]*-#{version_string}[a-f0-9]*, protocol version/, synergys_version)
+      assert_match(/synergyc [\-.0-9a-z]*-#{version_string}[a-f0-9]*, protocol version/, synergyc_version)
+    else
+      # The version information returned by synergys and synergyc is unusual.
+      # The actual version has four numbers (a.b.c.d), but the `--version` output
+      # omits the fourth number (i.e., it returns a.b.c). Additionally, the `--version`
+      # output appends some other numbers, letters, and symbols that are not easily
+      # recoverable, so we use regexps to deal with that.
+      version_string = Regexp.quote(version.to_s.split(".")[0..2].join("."))
+      assert_match(/synergys #{version_string}[\-.0-9a-z]*, protocol version/, synergys_version)
+      assert_match(/synergyc #{version_string}[\-.0-9a-z]*, protocol version/, synergyc_version)
+    end
+
+    # Spawn Synergy server.
+    IO.popen [
+      opt_bin/"synergys",
+      "-f", # run in foreground
+      "-c", conf_file,
+      "-a", "127.0.0.1:#{port}",
+      "-n", "test-server",
+      err: [:child, :out] # redirect child's stderr to its stdout
+    ] do |io_server|
+      # The whole point of the Synergy server is to read input from the user's
+      # mouse and keyboard. As a result, if we are running on MacOS and the
+      # user has not granted the "Accessibility" permission to the process
+      # running this test, the Synergy server will categorically refuse to
+      # start up. Unfortunately, there's nothing we can do about this, but
+      # it's not a flaw in Synergy itself (or in this test), so there's
+      # really nothing we can test except that the server threw the expected
+      # error message.
+      server_log = [log_trim.call(io_server.gets)]
+      permission_denied = (server_log[0] == LOG_PERMISSION_ERROR)
```

Comment:
So check for the error message then

Reply:
It does check for the error message. I've edited the comment to make this more clear.